Bio/PDB/ResidueDepth.py

```diff
@@ -528,19 +530,45 @@ def get_surface(model, MSMS="msms"):
             radius = _get_atom_radius(atom, rtype="united")
             pdb_to_xyzr.write(f"{x:6.3f}\t{y:6.3f}\t{z:6.3f}\t{radius:1.2f}\n")
 
-    # make surface
+    # Make surface
     surface_tmp = tempfile.mktemp()
-    MSMS = MSMS + " -probe_radius 1.5 -if %s -of %s > " + tempfile.mktemp()
+    msms_tmp = tempfile.mktemp()
+    MSMS = MSMS + " -probe_radius 1.5 -if %s -of %s > " + msms_tmp
     make_surface = MSMS % (xyz_tmp, surface_tmp)
     subprocess.call(make_surface, shell=True)
+    face_file = surface_tmp + ".face"
     surface_file = surface_tmp + ".vert"
     if not os.path.isfile(surface_file):
         raise RuntimeError(
-            f"Failed to generate surface file using command:\n{make_surface}"
+            "Failed to generate surface file using command:\n%s" % make_surface
         )
 
-    # read surface vertices from vertex file
+    # Read surface vertices from vertex file
     surface = _read_vertex_array(surface_file)
+
+    # Remove temporary files
+    if remove_tmp:
```

Comment:
I'd put these files in a list and iterate, to avoid the deep chain of try/except statements:

```
for fn in [xyz_tmp, surface_tmp, ...]:
    try:
        os.remove(fn)
    except OSError:
        pass

Reply:
@JoaoRodrigues @peterjc 
I believe my [latest commit](https://github.com/biopython/biopython/pull/3870/commits/53317680aaf6bdce5818f3793cf8a43e353e1dcb) addresses both of your concerns fully. If not, let me know, and I can make any remaining changes necessary.